### PR TITLE
Deprecate

### PR DESCRIPTION
**Pull request by ruslandoga**
Context: https://github.com/ExHammer/hammer/pull/104#issuecomment-2476141719

### CODE REVIEW

File: lib/hammer_plug.ex [hunk 1]
```diff
@@ -114,35 +114,72 @@ defmodule Hammer.Plug do
...
+    }
+
+    plug_name = plug_name(id_prefix)
+
+    Logger.warn("""
+    Hummer.Plug is deprecated. Please consider replacing it with a function plug:
+
+        plug :#{plug_name} # when action in ...
+
+        #{render_custom_plug(config) |> String.replace("\n", "\n    ")}
```
Comment (ruslandoga) [inline lib/hammer_plug.ex:167]: Example warning:

```
15:54:07.496 [warning] Hummer.Plug is deprecated. Please consider replacing it with a function plug:

    plug :rate_limit_video_upload # when action in ...

    defp rate_limit_video_upload(conn, _opts) do
      request_id = get_session(conn, :user_id)

      # note that request_id might be nil
      key = "video:upload:#{request_id}"
      scale = 60000
      limit = 10

      case Hammer.check_rate(key, scale, limit) do
        {:allow, _count} ->
          conn

        {:deny, _limit} ->
          conn |> send_resp(429, "Too Many Requests") |> halt()

        {:error, _reason} ->
          conn |> send_resp(429, "Too Many Requests") |> halt()
      end
    end
```

File: lib/hammer_plug.ex [hunk 2]
```diff
@@ -114,35 +114,72 @@ defmodule Hammer.Plug do
       #     end
 
   """
+  @behaviour Plug
   import Plug.Conn
+  require Logger
 
-  def init, do: init([])
-
+  @impl true
```
Comment (epinault) [inline lib/hammer_plug.ex:121]:
```suggestion
  @impl Plug
```

File: lib/hammer_plug.ex [hunk 3]
```diff
@@ -114,35 +114,72 @@ defmodule Hammer.Plug do
...
+
+        plug :#{plug_name} # when action in ...
+
+        #{render_custom_plug(config) |> String.replace("\n", "\n    ")}
+    """)
+
+    config
+  end
+
+  @impl true
```
Comment (epinault) [inline lib/hammer_plug.ex:173]:
```suggestion
  @impl Plug
```

**Comment by epinault** [comment]
@ruslandoga would it be helpful to chat in person ? I think we are misunderstanding each other you are misunderstanding my intention here

**Pull request closed without merging**